application-diagram-api/src/main/java/com/xwiki/diagram/internal/DiagramLinksListener.java

```diff
@@ -199,16 +178,11 @@ public Object doInHibernate(Session session) throws XWikiException
 
     /**
      * @return store system for execute store-specific actions.
+     * @throws ComponentLookupException
      * @throws DataMigrationException if the store could not be reached
      */
-    protected XWikiHibernateBaseStore getStore() throws DataMigrationException
+    private XWikiHibernateBaseStore getStore() throws ComponentLookupException
     {
-        try {
-            // The roleHint should be changed to XWikiHibernateBaseStore.HINT after updating to 9.11 parent.
-            return (XWikiHibernateBaseStore) this.componentManager.getInstance(XWikiStoreInterface.class, "hibernate");
-        } catch (ComponentLookupException e) {
-            throw new DataMigrationException(
-                String.format("Unable to reach the store for database %s", contextProvider.get().getWikiId()), e);
-        }
+        return (XWikiHibernateBaseStore) this.componentManager.getInstance(XWikiStoreInterface.class, "hibernate");
```

Comment:
You should inject the ``XWikiStoreInterface`` instead of using the component manager directly. If this creates problems because of a cycle in the event listener component dependencies then you can use a ``Provider<XWikiStoreInterface>``. You can keep this ``getStore()`` method to do the cast to ``XWikiHibernateBaseStore`` which you need to use.